### PR TITLE
Update doc for FinalClass

### PR DESCRIPTION
**Pull request by maheshkulkarni17**
**Update doc for FinalClass #7609**
Adding two examples to config_design.xml for FinalClass 

Example 1: 

<pre><font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ javac Test.java
<font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ java Test
OK
<font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ cat Test.java
class Parent{
Parent(){
}}
class Test {
public static void main(String args[]){
Parent s = new Parent();
System.out.println(&quot;OK&quot;);
}}//OK
</pre>

Example 2:
<pre><font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ javac Test.java
Test.java:6: error: Parent() has private access in Parent
Parent s = new Parent();
           ^
1 error
<font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ cat Test.java
class Parent{
private Parent(){
}}
class Test {
public static void main(String args[]){
Parent s = new Parent();
System.out.println(&quot;violation&quot;);
}}// violation, error: Parent() has private access in Parent
<font color="#8AE234"><b>sachin@sachin-Inspiron-5559</b></font>:<font color="#729FCF"><b>~/Desktop/java_programs</b></font>$ </pre>

i am beginner to this system , if i made any mistakes please guide me.

### CODE REVIEW

**Comment by rnveach** [comment]
> Test.java:6: error: Parent() has private access in Parent

Why are we adding an example that can't be compiled?

File: src/xdocs/config_design.xml [hunk 1]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+class Parent{
+private Parent(){
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+System.out.println("violation");
+}}// violation, error: Parent() has private access in Parent
+          
+        </p>
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: Please fix indentation

File: src/xdocs/config_design.xml [hunk 2]
```diff
@@ -290,7 +312,7 @@ public class FooTest {
             Checkstyle Style</a>
           </li>
         </ul>
-      </subsection>
+      </subsection>
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: No trailing space allowed

File: src/xdocs/config_design.xml [hunk 3]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+        <p>
+          Examples :
+          
+class Parent{
+Parent(){
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+System.out.println("OK");
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: Do not use print and okay, rather use `// ok`.

File: src/xdocs/config_design.xml [hunk 4]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+}} //OK
+          
+class Parent{
+private Parent(){
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+System.out.println("violation");
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: Do not use print, rather use `// violation`.
See other examples.

File: src/xdocs/config_design.xml [hunk 5]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+}}// violation, error: Parent() has private access in Parent
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: missing space after `}}`. Put the comment to line 297.

File: src/xdocs/config_design.xml [hunk 6]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+Parent(){
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+System.out.println("OK");
+}} //OK
+          
+class Parent{
+private Parent(){
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: missing space after `()`.

File: src/xdocs/config_design.xml [hunk 7]
```diff
@@ -277,6 +277,28 @@ public class FooTest {
...
+          Examples :
+          
+class Parent{
+Parent(){
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+System.out.println("OK");
+}} //OK
```
Comment (bmhm) [inline src/xdocs/config_design.xml]: lowercase `// ok`, also missing space.

**Comment by maheshkulkarni123OS** [comment]
@bmhm Made changes suggested by you , any further help is appreciated.
i am new to this community. Thank You..!!

**Comment by maheshkulkarni123OS** [comment]
@bmhm Made changes suggested by you , any further help is appreciated.
i am new to this community. Thank You..!!

File: src/xdocs/config_design.xml [hunk 8]
```diff
@@ -277,6 +277,26 @@ public class FooTest {
         <source>
 &lt;module name=&quot;FinalClass&quot;/&gt;
         </source>
+        <p>
+          Examples :
+          
+class Parent{
```
Comment (rnveach) [inline src/xdocs/config_design.xml:283]: Code should be surrounded by `<source>` tag.

File: src/xdocs/config_design.xml [hunk 9]
```diff
@@ -277,6 +277,26 @@ public class FooTest {
...
+Parent(){
```
Comment (rnveach) [inline src/xdocs/config_design.xml:284]: Code needs to be indented properly.

File: src/xdocs/config_design.xml [hunk 10]
```diff
@@ -277,6 +277,26 @@ public class FooTest {
...
+Parent s = new Parent();
+}} // ok
+          
+class Parent{
+private Parent() {
+}}
+class Test {
+public static void main(String args[]){
+Parent s = new Parent();
+}} // violation
```
Comment (rnveach) [inline src/xdocs/config_design.xml:297]: violation needs to be placed on exact line where violation will be printed.

File: src/xdocs/config_design.xml [hunk 11]
```diff
@@ -291,7 +311,7 @@ public class FooTest {
           </li>
         </ul>
       </subsection>
-
+
```
Comment (rnveach) [inline src/xdocs/config_design.xml:314]: Undo unnecessary change.

**Comment by romani** [comment]
@maheshkulkarni123OS , please finish PR.

File: src/xdocs/config_design.xml (hunk 8 above)
Comment (maheshkulkarni123OS) [reply src/xdocs/config_design.xml:282]: done

File: src/xdocs/config_design.xml (hunk 9 above)
Comment (maheshkulkarni123OS) [reply src/xdocs/config_design.xml]: done

**Comment by romani** [comment]
change is abandoned and incomplete.
